src/index.ts

```diff
@@ -315,7 +314,7 @@ export const Z = {
 
       static _parse(input: ParseInput): ParseReturnType<any> {
         const result = this._schema._parse(input);
-        if (isPromise(result)) {
+        if (result instanceof Promise) {
```

Comment:
Looks like is-promise uses duck typing.

https://github.com/then/is-promise/blob/master/index.js

Should we follow that practice or is instanceof going to work just fine? Maybe it won't work in older environments? Or perhaps there are situations where a promise is emulated?

Reply:
Hum i see what you mean you could also just, Nodes implementation does seem more robust. That being said it is a very simple implementation you could just copy paste nodes method and put into a utils.ts or something like that. You would achieve the same goal. I see from the https://github.com/sam-goodwin/zod-class/issues/25 that more people are having issues with this it would be good. 

I can only assume that everyone that wanted to use this in the frontent did wha i did, copied the code and put straight into the project. Ideally this would not be necessary

Reply:
I updated it to use the duck-typing method just to be 100% safe.